tools/build/ms-plugins.coffee

```diff
@@ -0,0 +1,113 @@
+{ walkFiles } = require('@resin.io/doxx-utils')
+HbHelper = require('@resin.io/doxx-handlebars-helper')
+
+path = require('path')
+fse = require('fs-extra')
+semver = require('semver')
+
+utils = require('../lib/utils')
+pythonHelper = require('../lib/python')
+
+
+removeFile = (obj, file, files) ->
+	if obj and obj is utils.UNSUPPORTED
+		delete files[file]
+
+copyFile = (src, dest) ->
+	try
+		fse.copySync(src, dest)
+	catch err
+		console.error(err)
+
+copyDebianFile = (obj, destDir) ->
+	copyFile('./supporting_files/debian.01_buildconfig', "#{destDir}/01_buildconfig")
+	copyFile('./supporting_files/debian.01_nodoc', "#{destDir}/01_nodoc")
+
+	# Systemd
+	if obj.$debian_suite.id is 'wheezy'
+		copyFile('./supporting_files/debian.entry-nosystemd.sh', "#{destDir}/entry.sh")
+	else
+		copyFile('./supporting_files/debian.entry.sh', "#{destDir}/entry.sh")
+		copyFile('./supporting_files/debian.launch.service', "#{destDir}/launch.service")
+
+	# QEMU
+	if obj.$debian_arch.id is 'aarch64'
+		copyFile('./qemu-aarch64-static', "#{destDir}/qemu-aarch64-static")
+	else if obj.$debian_arch.id != 'amd64' and obj.$debian_arch.id != 'i386'
+			copyFile('./qemu-arm-static', "#{destDir}/qemu-arm-static")
+
+copyAlpineFile = (obj, destDir) ->
+	copyFile('./supporting_files/alpine.rc.conf', "#{destDir}/rc.conf")
+	copyFile('./supporting_files/alpine.resin', "#{destDir}/resin")
+	copyFile('./supporting_files/alpine.entry.sh', "#{destDir}/entry.sh")
+	if obj.$alpine_arch.id != 'amd64' and obj.$alpine_arch.id != 'i386' 
+		copyFile('./qemu-arm-static', "#{destDir}/qemu-arm-static")
+
+copyDeviceFile = (obj, destDir) ->
+	if (/raspberrypi.*/i).test(obj.$device.id) and obj.$distro.id is 'debian'
+		copyFile('./supporting_files/device-base.resin-pinning', "#{destDir}/resin-pinning")
+
+copyGolangFile = (obj, destDir) ->
+	if obj.$language_variant.id != 'onbuild'
+		copyFile('./supporting_files/golang.go-wrapper', "#{destDir}/go-wrapper")
+
+copyPythonFile = (obj, destDir) ->
+	if obj.$device.id is 'edison' and obj.$distro.id is 'debian' and obj.$language_variant.id != 'onbuild'
+		copyFile('./supporting_files/python.setup.sh', "#{destDir}/setup.sh")
+
+checkWheezyNode = (obj, file, files) ->
+	type = file.split('/')[0]
+	if type is 'node'
+		if semver.eq(obj.$node_version.id, '6.3.1')
+			# Only generate dockerfile for debian wheezy node v6.3.1
+			obj.$isSupported = utils.UNSUPPORTED
+			if obj.$distro.id is 'debian' and obj.$language_variant.id is 'old_suite'
+				obj.$isSupported = utils.SUPPORTED
```

Comment:
The `SUPPORTED` value is never used (never checked and seems to be only needed to erase the `unsupported` flag. I'd drop its usage. Just do:

```
if semver.eq(obj.$node_version.id, '6.3.1')
  if not (obj.$distro.id is 'debian' and obj.$language_variant.id is 'old_suite')
    obj.$isSupported = utils.UNSUPPORTED
```

and drop it in other places. The `$isSupported` name is a bit confusing then, though (it already is in the next line anyway). So it'd better be `$isUnsupported` IMO.